ci/buildkite-pipeline.sh

```diff
@@ -303,6 +303,29 @@ pull_or_push_steps() {
     wait_step
   fi
 
+  # Version bump PRs are an edge case that can skip most of the CI steps
+  if affects .toml$ && affects .lock$ && ! affects_other_than .toml$ .lock$; then
```

Comment:
wdyt about something dumber where we check that `BUILDKITE_BUILD_AUTHOR_EMAIL` or `BUILDKITE_BUILD_AUTHOR` is the bot, and `BUILDKITE_COMMIT` looks like a version bump commit message

Reply:
I debated that (and I'm still open to it). 

For reference [here's the CI run for a recent version bump](https://buildkite.com/solana-labs/solana/builds/85913#0184e00b-71b1-4bf6-810b-f00b96367006).

I haven't dug into this but since we don't require signed commits I'm assuming some/most/all of the metadata can be spoofed. I didn't want to create an exception that someone could use to create a malicious PR that passes CI so they can DM a core contributor with "Hey GH is being weird about this version bump, can you try approving it?". 

But multiple safeguards have to fail for that attack vector to cause any harm and CI passing is only a minor signal that a PR is ok.

Reply:
the author check(s) part would be good either way

Reply:
`BUILDKITE_BUILD_AUTHOR_EMAIL` and `BUILDKITE_BUILD_AUTHOR` aren't currently populated. If you feel strongly about it I can look into populating them for the version bump PRs but based on the [Buildkite docs](https://buildkite.com/docs/pipelines/environment-variables) it might not be possible.